File: UBB/Fourth Semester/Projecting and Programming Methods/Laboratory 04/Lab4JAVA/src/main/java/domain/Child.java
package domain;

public class Child extends Person {
    int age;

    /**
     * constructor of object of type Child
     *
     * @param firstName - string
     * @param lastName  - string
     * @param age       - integer
     */
    public Child(String firstName, String lastName, int age) {
        super(firstName, lastName);
        this.age = age;
    }

    /**
     * getter for ageGroup
     *
     * @return - string, represents the age group for the child
     */
    public String getAgeGroup() {
        if (this.age >= 6 && this.age <= 8)
            return "6-8 years";
        if (this.age >= 9 && this.age <= 11)
            return "9-11 years";
        else
            return "12-15 years";
    }

    /**
     * getter methode for age
     *
     * @return - integer, represents the age of the child
     */
    public int getAge() {
        return age;
    }

    /**
     * setter methode for age
     *
     * @param age - integer , represents the age of the child
     */
    public void setAge(int age) {
        this.age = age;
    }
}

File: UBB/Fourth Semester/Projecting and Programming Methods/Laboratory 04/Lab4C#/AthleticsCompetition/domain/Child.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AthleticsCompetition.domain
{
    class Child : Person
    {
        /// <summary>
        /// getter and setter for child age
        /// </summary>
        public int Age { get; set; }

        /// <summary>
        /// constructor for child
        /// </summary>
        /// <param name="firstName"> string </param>
        /// <param name="lastName"> string </param>
        /// <param name="age"> integer </param>
        public Child(string firstName, string lastName, int age) : base(firstName, lastName)
        {
            this.Age = age;
        }

        /// <summary>
        /// gets the age group for a child
        /// </summary>
        /// <returns> string representing the group name </returns>
        public string GetAgeGroup()
        {
            if (this.Age >= 6 && this.Age <= 8)
                return "6-8 years";
            if (this.Age >= 9 && this.Age <= 11)
                return "9-11 years";
            else
                return "12-15 years";
        }
    }
}
